### ISSUE TITLE
[Request] For security reasons, framing is not allowed.

### ISSUE THREAD

**Issue by gaveitatry** [schomery/dictionary#52]
This popup is annoying.  Can you provide an option to disable it?  I don't want or need to see it every time that I translate a page, especially when the translated pages successfully load properly.  It's just an annoying prompt that doesn't serve any real purpose.  Makes my experience worse instead of better.

**Comment by gaveitatry** [comment]
Thanks to the developer of another extensions, I learned how to prevent the popups using this user script.

// ==UserScript==
// @name         DisableFraming Popups
// @match        https://translate.googleusercontent.com/*
// @match        https://www.translatoruser-int.com/*
// @grant        none
// @run-at       document-start
// ==/UserScript==

window.alert = function() {};

**Comment by schomery** [comment]
Can you share an example page with this issue? What is your browser?

**Comment by gaveitatry** [comment]
I tested it out on a random GitHub page that was in German.
https://github.com/Endor8/userChrome.js/tree/master/userChrome

I asked the developer of the Page Translator Revised web extension, and he replied:

> 
> 
> Yeah, there is code in https://github.githubassets.com/assets/github-bootstrap-e8bc289d.js, roughly like:
> 
> ```
> if (<loaded inside a frame>) {
>   alert("For security reasons, framing is not allowed.");
>   top.location.replace(document.location)
> }
> ```
> 
> Since the web versions of Google Translate and Microsoft Bing use frames they trigger this.
> 
> One solution is to do the translation inline on the page, like the Chrome browser [does](https://cs.chromium.org/chromium/src/components/translate/core/browser/translate_script.cc) and like the [old](https://github.com/jeremiahlee/page-translator) Page Translator extension does, but those won't pass AMO's review policies because they inject remote code. If you want to go down that road your best bet is to get a Bing Translate [API Key](https://www.microsoft.com/en-us/translator/business/trial/#get-started) and try to get Firefox's [built-in translation system](https://www.ghacks.net/2018/09/09/mozilla-working-on-google-translate-integration-in-firefox/) working. There are some API preferences to set, found [here](https://hg.mozilla.org/mozilla-central/file/tip/browser/components/translation/BingTranslator.jsm#l379).
> 
> But I think the easier solution is to remove the alert functionality using a TamperMonkey script (probably works in other *monkeys, but I haven't tried):
> 
> ```
> // ==UserScript==
> // @name         Disable framing popup
> // @match        https://translate.googleusercontent.com/*
> // @match        https://www.translatoruser-int.com/*
> // @grant        none
> // @run-at       document-start
> // ==/UserScript==
> 
> window.alert = function() {};
> ```



@schomery - By the way, the only reason that I am not using your web extension right now is because when I select "Translate with Bing" from the context menu, it translates the page on a new tab.  Can you add an option to make it so that we can translate using the same tab?

**Comment by schomery** [comment]
I need a page to test. I am asking this since we already have https://github.com/schomery/dictionary/blob/master/WebExtension/data/panel/inject.js#L126

**Comment by gaveitatry** [comment]
I am not sure what you mean.

I already gave you a page to test.

https://github.com/Endor8/userChrome.js/tree/master/userChrome

This particular GitHub page is in German.  In your web extension, and some others, if I translate this page in English using the Bing engine, I get a popup dialog that looks like this:

![image](https://user-images.githubusercontent.com/14252057/58195232-2127cc00-7c8d-11e9-8762-ff3266beb094.png)

I was able to get rid of this popup by creating a user script in VioletMonkey (or TamperMonkey or GreaseMonkey '''window.alert = function() {};'''

According to the other developer who gave me the user script, this popup can't be avoided because it is a part of Add-on Mozilla Org policies and restrictions.  Maybe he is right.   Maybe he is wrong.  If you figure out a way to make the extension to not make that popup without having to use a GreaseMonkey user script, that would be great.

I know you are showing me some window alert stuff in your extension, but it is not working on that German GitHub page that I tested.  According to the other developer, there is a way to stop the alert stuff from the extension code itself, but it involved injecting remote code with AMO won't let you do.  I don't understand code enough, so I'm not sure... but your code doesn't look like remote code, but it does say inject.js, so maybe it is remote.

I have no idea.  But I did give you a page to test.  Thanks for your other responses in my other threads.  I really look forward to your future update.
